src/library_manager/lib_manager.c

```diff
@@ -249,19 +326,26 @@ uint32_t lib_manager_allocate_module(const struct comp_driver *drv,
 	int ret;
 	uint32_t module_id = IPC4_MOD_ID(ipc_config->id);
 	uint32_t entry_index = LIB_MANAGER_GET_MODULE_INDEX(module_id);
+	struct lib_manager_mod_ctx *ctx = lib_manager_get_mod_ctx(module_id);
 
-	tr_dbg(&lib_manager_tr, "lib_manager_allocate_module(): mod_id: %#x",
-	       ipc_config->id);
+	tr_dbg(&lib_manager_tr, "lib_manager_allocate_module(): mod_id: %#x", module_id);
 
 	desc = lib_manager_get_library_module_desc(module_id);
-	if (!desc) {
+	if (!ctx || !desc) {
 		tr_err(&lib_manager_tr,
 		       "lib_manager_allocate_module(): failed to get module descriptor");
 		return 0;
 	}
 
 	mod = (struct sof_man_module *)((char *)desc + SOF_MAN_MODULE_OFFSET(entry_index));
 
+	for (unsigned int i = 0; i < ARRAY_SIZE(ctx->segment_size); i++) {
+		ctx->segment_size[i] = mod->segment[i].flags.r.length * PAGE_SZ;
+
+		if (i == SOF_MAN_SEGMENT_BSS && mod->type.load_type != SOF_MAN_MOD_TYPE_LLEXT)
+			ctx->segment_size[i] /= mod->instance_max_count;
```

Comment:
I would like move this if to `lib_manager_allocate_module_instance`

Reply:
@softwarecki I guess this could be done, and it wouldn't affect the LLEXT flow anyway, but I think it's more robust to have final sizes in this table and to be able to use them with no further modifications wherever needed?

Reply:
Is ctx->segment_size calculated value for module or is it size of loaded binary? Can you clarify it?

Reply:
I would prefer all bss-related calculations to be in one place. In my opinion, the name and location (in ctx) of this array suggests storing values read from the manifest. While reviewing the changes in the `lib_manager_allocate_module_instance` function, I was surprised that the division by the number of instances disappeared.

Reply:
> Is ctx->segment_size calculated value for module or is it size of loaded binary? Can you clarify it?

@pjdobrowolski that should be rather obvious from the code, sorry. It's calculated from section sizes of the loaded binary.

Reply:
> I would prefer all bss-related calculations to be in one place. In my opinion, the name and location (in ctx) of this array suggests storing values read from the manifest. While reviewing the changes in the `lib_manager_allocate_module_instance` function, I was surprised that the division by the number of instances disappeared.

@softwarecki currently that division is performed in two locations: in `lib_manager_allocate_module_instance()` and in `lib_manager_free_module_instance()`, after this change it's done only in one location, I think it's better.